LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/Regions/RegionLimits.cs

```diff
@@ -30,12 +32,19 @@ public RegionLimits((Hertz Min, Hertz Max) frequencyRange, HashSet<string> upstr
             this.startUpstreamDRIndex = startUpstreamDRIndex;
         }
 
-        public bool IsCurrentUpstreamDRValueWithinAcceptableValue(string dr) => this.upstreamValidDR.Contains(dr);
+        public bool IsCurrentUpstreamDRValueWithinAcceptableValue(string datr) =>
+            TryParseXpkDatr(datr) is { } dataRate && this.upstreamValidDR.Contains(dataRate);
 
-        public bool IsCurrentDownstreamDRValueWithinAcceptableValue(string dr) => this.downstreamValidDR.Contains(dr);
+        public bool IsCurrentDownstreamDRValueWithinAcceptableValue(string datr) =>
+            TryParseXpkDatr(datr) is { } dataRate && this.downstreamValidDR.Contains(dataRate);
 
-        public bool IsCurrentUpstreamDRIndexWithinAcceptableValue(ushort dr) => (dr >= this.startUpstreamDRIndex) && dr < this.startUpstreamDRIndex + this.upstreamValidDR.Count;
+        private static DataRate TryParseXpkDatr(string datr)
```

Comment:
Given the fact that this might be consciously null, what about enabling nullability for this class and adjusting the return type to DataRate?